rllib/utils/torch_utils.py

```diff
@@ -155,15 +155,26 @@ def clip_gradients(
         device = grads[0].device
 
         total_norm = torch.norm(
-            torch.stack([torch.norm(g.detach(), norm_type).to(device) for g in grads]),
+            torch.stack(
+                [
+                    torch.norm(g.detach(), norm_type)
+                    # Note, we want to avoid overflow in the norm computation, this does
+                    # not affect the gradients themselves as we clamp by multiplying and
+                    # not by overriding tensor values.
+                    .nan_to_num(neginf=-10e8, posinf=10e8).to(device)
+                    for g in grads
+                ]
+            ),
             norm_type,
-        )
+        ).nan_to_num(neginf=-10e8, posinf=10e8)
         if torch.logical_or(total_norm.isnan(), total_norm.isinf()):
             raise RuntimeError(
                 f"The total norm of order {norm_type} for gradients from "
                 "`parameters` is non-finite, so it cannot be clipped. "
             )
-        clip_coef = grad_clip / (total_norm + 1e-6)
+        clip_coef = grad_clip / torch.maximum(
+            torch.tensor(grad_clip), total_norm + 1e-6
```

Comment:
Two questions:
* Would this `torch.tensor()` pose a danger when on another device? GPU?
* Can we add a comment  here (or enhance the one below) explaining why we compute the coeff like this? What are the expected final values of the coeff (between 0.0 and 1.0)?

Reply:
Good catch! Yes, let's put the tensor on the `device` we have extracted before to be save. I will add then a note why we want the coefficient not larger than 1.0.